### PR TITLE
Add pre-commit and formatting tools

### PR DESCRIPTION
**Pull request by jgwerner**
- Adds Flake8 linter
- Adds black for code formatting
- Adds pre-commit config
- Adds Contributing guide

Related to #38 and #39

### CODE REVIEW

**Comment by jgwerner** [comment]
@yuvipanda as soon as you merge this PR I'll send one to refactor the existing LTI 1.1 setup based on #2 and then will add a separate one for LTI 1.3.

**Comment by jgwerner** [comment]
@yuvipanda I submitted #41 as a preview to this merge. I'll rebase once it's time to merge.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+pytest -v -k spawn ltiauthenticator/tests/test_lti.py
+```
+
+## Commit Messages, Changelog, and Releases
+
+This project includes `pre-commit` hooks configured to run `black`, `flake8`, and `pip-compile` before running and completing the `git commit ...` command. If these tools do not run when running the `git add ...` and `git commit ...` commands, make sure you install the pre-commit hooks with `pre-commit install`.
+
+### For Contributors
+
+This project uses Semantic Versioning with Conventional Commits to track major, minor, and patch releases. The `npm run release` command automates [CHANGLOG.md](./CHANGELOG.md) updates and release version metadata.
```
Comment (yuvipanda) [inline CONTRIBUTING.md:78]: I don't think Jupyter projects in general make a commitment to semantic versioning, so let's keep that out for now?

**Review by yuvipanda — CHANGES_REQUESTED** [review]
Thanks for doing this! One minor comment, otherwise happy to merge :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (jgwerner) [reply CONTRIBUTING.md]: Done.

**Comment by jgwerner** [comment]
@yuvipanda the comment regarding semantic versioning has been removed.

**Comment by yuvipanda** [comment]
w00t, thank you :)

**Pull request merged**